project/src/main/webapp/script.js

```diff
@@ -96,6 +154,128 @@ function tryAgain() {
   document.getElementById('place').innerHTML = '';
 }
 
+// Adds a search box to the map, and allows it to keep the user's updating location according to
+// his search box activity
+function addSearchBoxToMap(map, searchBoxElement) {
```

Comment:
Can this be split to several sub-methods?

Reply:
Did my best to make it more readable :)